src/llm/http_llm_calculator.cc

```diff
@@ -195,88 +193,81 @@ class OpenAIChatCompletionsRequest {
         it = this->doc.FindMember("repetition_penalty");
         if (it != this->doc.MemberEnd()) {
             if (!it->value.IsDouble())
-                return false;
+                return absl::InvalidArgumentError("repetition_penalty is not a double");
             this->repetitionPenalty = it->value.GetDouble();
-            // TODO: Validate?
         }
 
         // diversity_penalty: float; optional - defaults to 1.0
         // Extension, unsupported by OpenAI API and vLLM, however available in CB lib
         it = this->doc.FindMember("diversity_penalty");
         if (it != this->doc.MemberEnd()) {
             if (!it->value.IsDouble())
-                return false;
+                return absl::InvalidArgumentError("diversity_penalty is not a double");
             this->diversityPenalty = it->value.GetDouble();
-            // TODO: Validate?
         }
 
         // length_penalty: float; optional - defaults to 1.0
         // Extension, unsupported by OpenAI API however supported by vLLM and CB lib
         it = this->doc.FindMember("length_penalty");
         if (it != this->doc.MemberEnd()) {
             if (!it->value.IsDouble())
-                return false;
+                return absl::InvalidArgumentError("length_penalty is not a double");
```

Comment:
I think we should go with a "floating point number" instead of a "double" since it's a more general term, better suiting for the API I believe.